quinn/src/endpoint.rs

```diff
@@ -523,7 +541,12 @@ impl State {
                                 .send(ConnectionEvent::Proto(event));
                         }
                     }
-                    Transmit(t) => self.outgoing.push_back(udp_transmit(t)),
+                    Transmit(t) => {
+                        let contents_len = t.contents.len();
+                        self.outgoing.push_back(udp_transmit(t));
+                        self.transmit_queue_contents_len
+                            .fetch_add(contents_len, Ordering::Relaxed);
```

Comment:
This only counts packets after they've been received by the connection, but they still occupy space while waiting in the `self.events` queue. Is that intended?

Reply:
Yes, in my test, I have not seen a build-up of items in the events  it is always the outgoing queue the problem. I think the difference here is the send-limiter when draining down the outgoing queue.

Reply:
Ah, good to know. We should document this in a comment, since it's not obvious otherwise. That also means there's no need for the counter to be an atomic, since we always have `&mut self` when accessing it; let's use a plain `usize` instead.